### ISSUE TITLE
[2.x] IllegalStateException on Observable.timer with 0 milliseconds

### ISSUE THREAD

**Issue by nhaarman** [ReactiveX/RxJava#4926]
- Library version 2.0.3

The following test occasionally fails (about 200 out of 1000 fail):

```kotlin
@Test
fun test() {
    var t: Throwable? = null
    RxJavaPlugins.setErrorHandler { t = it }
    Observable.timer(0L, MILLISECONDS).blockingFirst()
    t?.let { throw it }
}
```

Stacktrace:

```
java.lang.IllegalStateException: Disposable already set!

	at io.reactivex.internal.disposables.DisposableHelper.reportDisposableSet(DisposableHelper.java:144)
	at io.reactivex.internal.disposables.DisposableHelper.setOnce(DisposableHelper.java:70)
	at io.reactivex.internal.operators.observable.ObservableTimer$IntervalOnceObserver.setResource(ObservableTimer.java:74)
	at io.reactivex.internal.operators.observable.ObservableTimer.subscribeActual(ObservableTimer.java:40)
	at io.reactivex.Observable.subscribe(Observable.java:10514)
	at io.reactivex.Observable.blockingFirst(Observable.java:4535)
	at myapp.MyTest.test(MyTest.kt:15)
```

Whenever the test fails, [IntervalOnceObserver#setResource](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/internal/operators/observable/ObservableTimer.java#L40) is called _after_ it is disposed.

**Comment by akarnokd** [comment]
Thanks for reporting. If the emission happens before the `setResource` call, the `setResource` will find the wrong reference there and complain about it. I'll fix it shortly.

**Comment by akarnokd** [comment]
Closing via #4927

**Comment by guidedways** [comment]
Tried v2.0.2 in production and seeing a lot of these crashes come up. I know this has been fixed, but do we have an ETA for v2.0.4? Thank you.

**Comment by JakeWharton** [comment]
See https://github.com/ReactiveX/RxJava/issues/4947
